src/internal/components/tooltip/index.tsx

```diff
@@ -7,27 +7,36 @@ import { Transition } from '../transition';
 import PopoverContainer from '../../../popover/container';
 import PopoverBody from '../../../popover/body';
 import Portal from '../portal';
+import { PopoverProps } from '../../../popover/interfaces';
```

Comment:
Depending internal component on public feels wrong.

How about moving the type here and re-export it in popover? See [`OptionDefinition`](https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20optiondefinition&type=code) for a similar treatment

Reply:
We use public component `Popover` inside the internal and the appropriate interface for it. How it could be beneficial to duplicate the props?

Reply:
This looks like a circular dependency, popover extends the tooltip, so all dependencies between these two components should be one direction

Reply:
It would be a circular dependency if the `Tooltip` component was used within the `Popover`, which is not a case. Nevertheless, moved the type in-place:

```javascript
position?: 'top' | 'right' | 'bottom' | 'left';
```

Reply:
I see that there were popover imports in this file before, but it is still good to not exacerbate the issue.

You may also want to talk with other team members about possible refactoring here